backend/src/transactions/transactions.controller.ts

```diff
@@ -1,15 +1,59 @@
-import { Controller, Body, Post } from '@nestjs/common'
-import { CreateTransactionDto } from './dto/transaction.dto'
-import { TransactionsService } from './transactions.service'
+import { BadRequestException, Body, Controller, Delete, Get, Param, ParseIntPipe, Patch, Post, UseGuards, UsePipes, ValidationPipe } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { GetUser } from '../auth/get-user.decorator';
+
+import { User } from '../users/user.entity';
+import { CreateTransactionDto } from './dto/transaction.dto';
+import { Transaction } from './transaction.entity';
+import { TransactionsService } from './transactions.service';
 
 @Controller('transactions')
+@UseGuards(AuthGuard())
 export class TransactionsController {
-  constructor(private transactionService: TransactionsService) {
+  constructor(private transactionsService: TransactionsService) {
+  }
+
+  @Get()
+  getTransactions(
+    @GetUser() user: User
+  ): Promise<Transaction[]> {
+    return this.transactionsService.getTransactions(user);
+  }
+
+  @Get('/:id')
+  getTransactionById(
+    @Param('id', ParseIntPipe) id: number,
+    @GetUser() user: User
+  ): Promise<Transaction> {
+    return this.transactionsService.getTransactionById(id, user);
   }
-  @Post('/')
-  create(@Body() createTransactionDto: CreateTransactionDto) {
-    // console.log(createTransactionDto)
-    // this.transactionService.create(createTransactionDto);
-    return 'Transaction created'
+
+  @Post()
+  @UsePipes(ValidationPipe)
+  createTransaction(
+    @Body() createTransactionDto: CreateTransactionDto,
+    @GetUser() user: User
+  ): Promise<Transaction> {
+    if (!createTransactionDto.amount) {
+      throw new BadRequestException();
+    }
```

Comment:
Please use [ValidationPipes](https://docs.nestjs.com/techniques/validation#transform-payload-objects) to check for payload